docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/data-import/spark-writer.md

```diff
@@ -0,0 +1,344 @@
+
+#### 3.3.2 Nebula Configuration
+
+| Field | Default Value | Required | Description |
+| --- | --- | --- | --- |
+| nebula.addresses | / | yes | query engine IP list, separated with comma |
+| nebula.user | / | yes | user name |
+| nebula.pswd | / | yes | password |
+| nebula.space | / | yes | space to import data |
+| nebula.connection.timeout | 3000 | no | Thrift timeout |
+| nebula.connection.retry | 3 | no | Thrift retry times |
+| nebula.execution.retry | 3 | no | nGQL execution retry times |
+
+#### 3.3.3 Mapping of Tags and Edges
+
+The options for tag and edge mapping are very similar. The following describes the same options first, and then introduces the unique options of `tag mapping` and `edge mapping`.
+
+* **Same Options**
+  * `type` is a case insensitive required field that specifies data type in the context, currently supports Parquet, JSON, ORC and CSV
+  * `path` is applied to HDFS data source and specifies the absolute path of HDFS file or directory, it's a required field when the type is HDFS
+  * `exec` is applied to Hive data source, it's a required filed when the query type is HIVE
+  * `fields` is a required filed that maps the columns of the data source to properties of tag / edge
+
+* **unique options for tag mapping**
+  * `vertex` is a required field that specifies a column as the vertex ID column
+* **unique options for edge mapping**
+  * `source` is a required field that specifies a column in the input source as the **source vertex** ID column
+  * `target` is a required field that specifies a column as the **dest vertex** ID column
+  * `ranking` is an optional field that specifies a column as the edge ranking column when the inserted edge has a ranking value
+
+#### 3.3.4 Data Source Mapping
+
+* **HDFS Parquet Files**
+  * `type` specifies the input source type. When it is parquet, it's a case insensitive required field
+  * `path` specifies the HDFS file directory, it's a required field that must be the absolute directory
+* **HDFS JSON Files**
+  * `type` specifies the type of the input source. When it is JSON, it's a case insensitive required field
+  * `path` specifies the HDFS file directory, it's a required field that must be absolute directory
+* **HIVE ORC Files**
+  * `type` specifies the input source type. When it is ORC, it's a case insensitive required field
+  * `path` specifies the HDFS file directory, it's a required field that must be absolute directory
+* **HIVE CSV Files**
+  * `type` specifies the input source type. When it is CSV, it's a case insensitive required field
+  * `path` specifies the HDFS file directory, it's a required field that must be the absolute directory
+* **HIVE**
+  * `type` specifies the input source type. When it is HIVE, it's a case insensitive required field
+  * `exec` is a required field that specifies the HIVE executed query
+
+### 3.4 Import Data
+
+Input data use the following command:
```

Comment:
The input data uses the ...

Reply:
I'd rather an imperative form here. Changed to 
Input data with the following command: